h/services/bulk_annotation.py

```diff
@@ -82,51 +78,44 @@ def annotation_search(
         authority: str,
         audience: dict,
         updated: dict,
-        fields: list = None,
         limit=100000,
-    ) -> Union[List[Annotation], Row]:
+    ) -> List[BulkAnnotation]:
         """
         Get a list of annotations or rows viewable by an audience of users.
 
-        Using a fields argument will switch the return type from annotation
-        objects to row objects. This is more efficient if you only need a
-        few fields but less convenient if you want many fields. Check the
-        `_FIELDS` attribute for acceptable fields and names.
-
         :param authority: The authority to search by
         :param audience: A specification of how to find the users. e.g.
             {"username": [...]}
         :param updated: A specification of how to filter the updated date. e.g.
             {"gt": "2019-01-20", "lte": "2019-01-21"}
-        :param fields: A list of string descriptions of fields to return
-            instead of full annotation objects.
         :param limit: A limit of results to generate
 
         :raises BadDateFilter: For poorly specified date conditions
-        :raises BadFieldSpec: For poorly specified fields
         """
 
         results = self._db.execute(
-            self._search_query(
-                authority, audience=audience, updated=updated, fields=fields
-            ).limit(limit)
+            self._search_query(authority, audience=audience, updated=updated).limit(
+                limit
+            )
         )
 
-        if fields is None:
-            results = results.scalars()
-
-        return results.all()
+        return [
+            BulkAnnotation(
+                username=row.username, authority_provided_id=row.authority_provided_id
+            )
+            for row in results.all()
+        ]
 
     @classmethod
-    def _search_query(cls, authority, audience, updated, fields=None) -> Select:
+    def _search_query(cls, authority, audience, updated) -> Select:
         """Generate a query which can then be executed to find annotations."""
 
-        if fields is None:
-            query = sa.select(Annotation)
-        else:
-            query = sa.select(list(cls._field_names_to_columns(fields))).select_from(
-                Annotation
-            )
+        columns = [
+            cls._AUTHOR.username,
+            Group.authority_provided_id,
+        ]
+
+        query = sa.select(columns).select_from(Annotation)
```

Comment:
Could maybe just inline `columns`, it's only used in one place

```suggestion
        query = sa.select([cls._AUTHOR.username, Group.authority_provided_id]).select_from(Annotation)
```